src/include/sof/audio/module_adapter/intel/core/kernel/logger/log.h

```diff
@@ -0,0 +1,414 @@
+ * derivatives also FLOG, ALOG, SLOG etc) in code will:
+ * 1. [COMPILATION] Create log_entry in .static_log section.
+ * 2. [COMPILATION] Check message format (should be printf-like).
+ * 3. [RUNTIME] Check whether message can be logged.
+ */
+
+#define LOGGER_THREADPTR_OFFSET 120
+
+#ifdef MODULE_NAME
+#define LOGGER_LIB_ID_VAR_NAME CONCAT(CONCAT(__, MODULE_NAME), _lib_id)
+extern const volatile uint32_t LOGGER_LIB_ID_VAR_NAME;
+#define LOGGER_LIB_ID_WRAP(input) (((uint32_t)input) | (LOGGER_LIB_ID_VAR_NAME << 3))
+#else
+#define LOGGER_LIB_ID_WRAP(input) input
+#endif
+
+/*! The following function pointers are used to put logs in different ways */
+/*! depending on selected logging backend */
+typedef void (*log_put_0_fn)(void*, uint32_t);
+typedef void (*log_put_1_fn)(void*, uint32_t, uint32_t);
+typedef void (*log_put_2_fn)(void*, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_3_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_4_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_5_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t);
+typedef void (*log_put_6_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t,
+				uint32_t);
+typedef void (*log_put_7_fn)(void*, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t,
+				uint32_t, uint32_t);
+typedef void (*log_timer_callback_fn)(void *param);
+
+/*! Logger descriptor. */
+typedef struct logger {
+	struct log_transport *transport;
+	uint32_t log_en_mask_;
+	/*! pointers to functions to put logs */
+	/*! single function for different param count */
+	log_put_0_fn log_put_0;
+	log_put_1_fn log_put_1;
+	log_put_2_fn log_put_2;
+	log_put_3_fn log_put_3;
+	log_put_4_fn log_put_4;
+	log_put_5_fn log_put_5;
+	log_put_6_fn log_put_6;
+	log_put_7_fn log_put_7;
+
+	/*! amount of dropped messages */
+	uint32_t dropped_counter_;
+	/*! reference host time */
+	uint64_t host_time_reference_;
+	/*! reference dsp time*/
+	uint64_t dsp_time_reference_;
+
+	/*! Pointer to log buffer descriptor */
+	struct log_buffer *log_buffer;
+	/*! ACTUAL log buffer data capacity not log buffer size! */
+	uint32_t buffer_capacity;
+
+	uint32_t cached_last_ro_f;
+	uint32_t cached_wo;
+} logger_s;
+
+/*! Log level priority enumeration. */
+typedef enum log_priority {
+	/*! Critical message. */
+	L_CRITICAL = 0,
+	/*! Error message. */
+	L_ERROR = 0,
+	/*! High importance log level. */
+	L_HIGH = 1,
+	/*! Warning message. */
+	L_WARNING = 1,
+	/*! Medium importance log level. */
+	L_MEDIUM = 2,
+	/*! Low importance log level. */
+	L_LOW = 3,
+	/*! Information. */
+	L_INFO = 3,
+	/*! Verbose message. */
+	L_VERBOSE = 4,
+	L_DEBUG   = 4,
+	L_MAX     = 5,
+} AdspLogPriority,
+	log_priority_e;
+
+/*! Type of log source	*/
+typedef enum log_source {
+	L_INFRA      = 0,
+	L_HAL        = 1,
+	L_MODULE     = 2,
+	L_AUDIO      = 3,
+	L_SENSING    = 4,
+	L_ULP_INFRA  = 5,
+	L_ULP_MODULE = 6,
+	L_VISION     = 7,
```

Comment:
Same here. Library was build with those definitions. It will generate log entry with L_MODULE type. In the SystemServiceLogMessage() routine body we can generate Zephyr log descriptor but first we need to decode what was obtained from the library.

Reply:
I think the Zephyr LOG classes are static objects, i.e. they need to be declared at build time. @andyross ? e.g. from zephyr_domain.c
```c
LOG_MODULE_DECLARE(ll_schedule, CONFIG_SOF_LOG_LEVEL);
```
We maybe able to dynamically create them though ?

Reply:
My understanding is no, actually.  The log modules are fixed (they go with C translation units and are implemented with linker section names, there's no runtime data structure).  Log backends and filtering state can be changed at runtime, but the set of modules is assumed to be static.  (I think -- not really an expert on logging).

If there's need for having loadable modules integrate with log module identifiers, there may need to be some infrastructure work done.  @dcpleung or @nordic-krch might have ideas.